docs/data-sources/hardware_reservation.md

```diff
@@ -0,0 +1,36 @@
+---
+page_title: "Equinix Metal: metal_hardware_reservation"
+subcategory: ""
+description: |-
+  Retrieve Equinix Metal Hardware Reservation
+---
+
+# metal_hardware_reservation
+
+Use this data source to retrieve a [hardware reservation resource from Equinix Metal](https://metal.equinix.com/developers/docs/deploy/reserved/).
+
+## Example Usage
+
+```hcl
+data "hardware_reservation" "example" {
+  id     = "4347e805-eb46-4699-9eb9-5c116e6a0172"
+}
+```
+
+## Argument Reference
+
+* `id` - (Required) ID of the hardware reservation
```

Comment:
`device_id` may be the most helpful lookup after `id`. We can add that in a separate PR (we'll need an issue to track).